interfaces/java/CMakeLists.txt

```diff
@@ -7,143 +7,289 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-if(HELICS_SWIG_GENERATE_INTERFACE_FILES_ONLY)
-    include(${CMAKE_CURRENT_SOURCE_DIR}/javaSwigGenerateOnly.cmake)
+cmake_minimum_required(VERSION 3.4)
+project(helicsJAVA)
+
+get_directory_property(hasParent PARENT_DIRECTORY)
+
+if(NOT hasParent)
+    message(STATUS "Standalone Java interface build")
+
+    if(CMAKE_VERSION VERSION_LESS 3.16)
+        cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
+    else()
+        cmake_policy(VERSION 3.16)
+    endif()
+
+    # Setup options
+    option(HELICS_ENABLE_SWIG "Use SWIG to generate the Java interface files" ON)
+
+    include(GNUInstallDirs)
+
+    # Setup rpath
+    if(NOT DEFINED CMAKE_MACOSX_RPATH)
+        set(CMAKE_MACOSX_RPATH ON)
+    endif()
+
+    if(NOT DEFINED CMAKE_INSTALL_RPATH_USE_LINK_PATH)
+
+        set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
+    endif()
+
+    # Add the local directory to the rpath
+    if(NOT APPLE)
+        list(APPEND CMAKE_INSTALL_RPATH $ORIGIN)
+        list(APPEND CMAKE_INSTALL_RPATH
+            "\$ORIGIN/../${CMAKE_INSTALL_LIBDIR}"
+        )
+    else()
```

Comment:
what does this do on Windows?

Reply:
Nothing, Windows doesn't support rpaths.